package/kedro_viz/server.py

```diff
@@ -210,13 +218,16 @@ def pretty_name(name):
                 "tags": sorted(node.tags),
             }
         )
-        all_tags.update(node.tags)
+
         for data_set in node.inputs:
             namespace = data_set.split("@")[0]
+            namespace_to_layer[namespace] = data_set_to_layer_map.get(data_set)
             edges.append({"source": _hash(namespace), "target": task_id})
             namespace_tags[namespace].update(node.tags)
+
         for data_set in node.outputs:
             namespace = data_set.split("@")[0]
+            namespace_to_layer[namespace] = data_set_to_layer_map.get(data_set)
```

Comment:
Just checking I understood correctly, so if a user assigns to different layers to a transcoded dataset (A@spark in intermediary and A@pandas - used later - in primary), is it the output one overwrites the existing entry in `namespace_to_layer`? Or sth else happens? Are they warned anywhere that they're doing sth weird with their layers for that namespace (A), or is that part of future work?

Reply:
@lorenabalan As I understood, `A` should be the name of a dataset, isn't it? So so we expect to have `A@spark` and `A@pandas` in one file.
I just took a look at unit tests and found the following examples: "fred_in@pandas" and "fred_out@pandas".

So maybe I don't understand the transcoded dataset correctly.

Reply:
So in your catalog you'd have
```yaml
A@spark:
    type: spark.SparkDataSet
    layer: de
A@pandas:
    type: pandas.PandasDataSet
    layer: ds   # bad
```
and my understanding was that, as far as the catalog knows, in `self._data_sets` you'd have both keys "A@pandas" and "A@spark".

Reply:
So depending on how inputs/outputs are set, you could end up with `namespace_to_layer` to look like {"A": "de"} or {"A": "ds"}. Please correct me if I'm missing sth!

Reply:
I think that might be a product question... Does kedro-viz currently visualise transcoded dataset?

Reply:
I think it does. I remember Gordon doing some hackery to show that it was a transcoded dataset.

Reply:
In Viz you only see "A", the two get merged into one dataset. Anything that's after @ is irrelevant. But in this case if we attach one layer over the other, it's not clear which wins (to the user). It's bad behaviour to start with anyway, to assign 2 different layers, just wondering if/how we thought about this or if this is surfaced to the user somewhere yet.

Reply:
@lorenabalan As you said kedro-viz will get two `A` identical nodes that according to your example related to different values (in one case it is `pandas` data in another `spark`). How pandas and spark data be merged?

Reply:
Let's imaging we have the following:
DE pipeline: `data1@spark` -> `node1` -> `data1@pandas`.
How it will be depicted on Kedro-Viz?

Reply:
<img width="287" alt="Screen Shot 2020-04-21 at 15 45 30" src="https://user-images.githubusercontent.com/24272541/79867541-2d5d6180-83e7-11ea-9dae-746b12e97b21.png">
```
def create_pipeline(**kwargs):
    return Pipeline(
        [
            node(
                report_accuracy,
                "spark_data_2@user",
                "spark_data_2@custom"
            )
        ]
    )
```

Is this the correct picture?

Reply:
Where catalog.yml is:

```
spark_data_2@user:
  type: spark.SparkDataSet
  filepath: /tmp/spark/test_data

spark_data_2@custom:
  type: pandas.CSVDataSet
  filepath: /tmp/spark/test_data/test.csv
```

Reply:
So your example pipeline is not allowed, Kedro will complain when user is trying to write to the same thing you're reading from, i.e. the same `spark_data_2` "namespace".

Reply:
I have another use case which we can discuss online.
I've got confused because from kedro-viz perspective I get two identical nodes.

Reply:
I think unless we start displaying the full transcoded name in viz, there is no way at the moment we can show transcoded datasets in 2 different layers. Probably worth putting it in a ticket or a TODO comment.